### ISSUE TITLE
RTX5000 GPU Support?

### ISSUE THREAD

**Issue by holycowdude** [kohya-ss/musubi-tuner#235]
_Thank you for your amazing software, i've trained some hunyuan video loras with your software and they work very well!_

Lately i've been using the superb [musubi-tuner-scripts](https://github.com/sdbds/musubi-tuner-scripts) UV version as its really easy to install but after getting an RTX5090 GPU I cant get it to work

Is there a way to get the UV version working with an RTX5090 please or should I pip based installation (if so how do i get it to work with the rtx5090 - do i need to use a nightly cuda / pytorch?)

Thanks!

**Comment by df2df** [comment]
Does this have 5xxx support in windows? 

With my 5070 I could get flux training to work in kohya with AdamW (but not AdamW8bit) but in musubi nothing works and it stays stuck at step 0 with no further indication it's doing anything regardless of optimzer

**Comment by maybleMyers** [comment]
Download https://github.com/maybleMyers/H1111/blob/main/requirementsTorch27.txt
put it in the musubi-tuner root folder
then 
python -m venv env
env/scripts/activate
pip install typing-extensions
pip install torch==2.7.0+cu128 torchvision==0.22.0+cu128 --index-url https://download.pytorch.org/whl/cu128
pip install -r requirementsTorch27.txt

I think this will work, because it works for my fork.

**Comment by AlonDan** [comment]
Can you please help with the missing things that always clash with each other on Windows?

What about Triton + Sageattention specific versions so they ALL fit to the specific Python 3.10 (or other version) for the Cuda 12.8 
or anything missing?

If anyone can approve it helps and make a complete step-by-step installation guide for Musubi-Tuner on Windows with RTX 5000 series it will be great!

Thanks ahead!

**Comment by Santelomar** [comment]
I'm also struggling with installing it on Windows 11 + Nvidia RTX 5090
What's the difference between all these options: Xformers / FlashAttention  / SageAttention

and what's the best for training Wan 2.1 image to video Loras on Windows 11 + RTX 5090 ?

Maybe if somebody with RTX 50xx could make a portable version specific for Windows Musubi-Tuner and most up to date of these hard to install things it will be super helpful and appreciated, obviously without the models.

**Comment by rwfsmith** [comment]
These are all of the packages that I have installed and it works for my 5090 on Windows 11, at least.

(venv) G:\Projects\musubi-tuner>pip list
Package                 Version      Editable project location
----------------------- ------------ --------------------------------------
absl-py                 2.2.2
accelerate              1.6.0
ascii-magic             2.3.0
av                      14.0.1
bitsandbytes            0.45.4
certifi                 2025.4.26
charset-normalizer      3.4.1
colorama                0.4.6
contourpy               1.3.2
cycler                  0.12.1
diffusers               0.32.1
easydict                1.13
einops                  0.7.0
filelock                3.13.1
flash_attn              2.7.4.post1
fonttools               4.57.0
fsspec                  2024.6.1
ftfy                    6.3.1
grpcio                  1.71.0
huggingface-hub         0.30.0
idna                    3.10
importlib_metadata      8.7.0
Jinja2                  3.1.4
kiwisolver              1.4.8
Markdown                3.8
MarkupSafe              2.1.5
matplotlib              3.10.1
mpmath                  1.3.0
networkx                3.3
ninja                   1.11.1.4
numpy                   2.1.2
opencv-python           4.10.0.84
packaging               25.0
pillow                  11.0.0
pip                     25.1
protobuf                6.30.2
psutil                  7.0.0
pyparsing               3.2.3
python-dateutil         2.9.0.post0
PyYAML                  6.0.2
regex                   2024.11.6
requests                2.32.3
safetensors             0.4.5
sageattention           2.1.1        G:\Projects\musubi-tuner\SageAttention
setuptools              70.2.0
six                     1.17.0
sympy                   1.13.3
tensorboard             2.19.0
tensorboard-data-server 0.7.2
tokenizers              0.20.3
toml                    0.10.2
torch                   2.7.0+cu128
torchaudio              2.7.0+cu128
torchvision             0.22.0+cu128
tqdm                    4.67.1
transformers            4.46.3
triton-windows          3.3.0.post19
typing_extensions       4.12.2
urllib3                 2.4.0
voluptuous              0.15.2
wcwidth                 0.2.13
Werkzeug                3.1.3
wheel                   0.45.1
xformers                0.0.30
zipp                    3.21.0

**Comment by Santelomar** [comment]
Thanks for sharing @rwfsmith 
It seems like you already know what you're doing and you have RTX 5090 as well ❤️
I hope you can answer some of these questions for everyone else to get start with 50xx series, and in general Wan tips & tricks will be very appreciated.

Regarding the list you shared, just to make sure if I'm doing it correct:
- All I need to do is to copy past all to:  `requirements_5090.txt` (for example)
- and then from my main musubi-tuner folder, after activate venv:
`pip install -r requirements_5090.txt `

- This will also take care of all the "problematic" Triton, SageAttention and the others in one take?


- I guess I should install from scratch, because I currently used Python 3.10 since I had issues with 3.12

Probably with your list of specific versions it works perfectly fine for 5090 as you mentioned.

---

- Did you manage to train Wan 2.1 i2v ? 🤔
- If so, can you please explain how to setup the dataset? 
- is it .mp4 files? .txt file for each?
- or is it based on extracting frames first into folders? each folder is a video?

- For very first test training just to make sure it works, any suggestions?

Of course, I'll try the list you shared first (if I did it correct), but then I have no clue how to even start the Docs seems overwhelming, 
I just want to test something quick to make sure my PC can train without errors.


Thanks ahead for any tips & tricks, and for the dedicated 5090 list!

**Comment by rwfsmith** [comment]
For Sage Attention do this:

```
git clone https://github.com/thu-ml/SageAttention.git
cd sageattention 
pip install -e .

```

SageAttention is only used for inference, though, so doesn't help with training. I think you can use FlashAttention or Xformers to speed up training?

This was my first lora and it wasn't great, so take this all with a grain of salt, but I focused mostly on training t2v, but it should work similarly for i2v. I used both images and videos, took some 2-4 second clips in .mp4 format, put them into separate folders.

this is my dataset.toml, the first dataset is images, second is video. I think for i2v you just use videos? I was going for more of a style lora than a motion lora. You could use lower resolution videos for that, I've seen people say 256x256 is good enough.

for repeating motions you could also change the frame_extraction to head and set which frames you want to extract, I think? 

something like:
target_frames = [1, 25, 45]
frame_extraction = "head"

dataset.toml:
# resolution, caption_extension, batch_size, num_repeats, enable_bucket, bucket_no_upscale should be set in either general or datasets
# otherwise, the default values will be used for each item

# general configurations
[general]
caption_extension = ".txt"
batch_size = 1
enable_bucket = true
bucket_no_upscale = false

[[datasets]]
resolution = [960, 544]
image_directory = "g:/monogatari images/dataset2/images"
cache_directory = "g:/monogatari images/dataset2/images/cache"
num_repeats = 1 # optional, default is 1. Number of times to repeat the dataset. Useful to balance the multiple datasets with different sizes.

[[datasets]]
resolution = [512, 288]
video_directory = "g:/monogatari images/dataset2/clips"
cache_directory = "g:/monogatari images/dataset2/clips/cache"
frame_extraction = "full"
max_frames = 96
source_fps = 23.976204

# other datasets can be added here. each dataset can have different configurations

**Comment by rwfsmith** [comment]
to do the training, I ran all of this in order:

create cache:

```
python wan_cache_latents.py --dataset_config "G:/monogatari images/dataset2/dataset.toml" --vae "./Wan2.1_VAE.pth"

python wan_cache_text_encoder_outputs.py --dataset_config "G:/monogatari images/dataset2/dataset.toml" --t5 "./models_t5_umt5-xxl-enc-bf16.pth" --batch_size 16 
```

train lora:

`accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 wan_train_network.py --task t2v-14B --dit ./wan2.1_t2v_14B_bf16.safetensors --dataset_config "G:/monogatari images/dataset2/dataset.toml" --mixed_precision bf16 --optimizer_type adamw8bit --learning_rate 2e-4 --gradient_checkpointing --max_data_loader_n_workers 2 --persistent_data_loader_workers  --network_module networks.lora_wan --network_dim 32 --timestep_sampling shift --discrete_flow_shift 3.0 --max_train_epochs 100 --save_every_n_epochs 10 --seed 42 --output_dir "G:/monogatari images/lora-v1.1-14b" --output_name wan_14b_monogatari --flash_attn --network_args "loraplus_lr_ratio=4" --fp8`

convert lora to be usable in comfyui:

`python convert_lora.py --input "G:\monogatari images\lora2\wan_1.3b_monogatari.safetensors" --output .\wan_t2v_14B_monogatari_anime_style.safetensors --target other
`

You can start with fewer --max_train_epochs to just 10 to see if it works. As you train more epochs, the average loss decreases. The lower the better, generally, but lower doesn't guarantee good results. Good tagging and data is the most important part.

**Comment by AlonDan** [comment]
Your help is pure gold @rwfsmith ❤️
Thanks for sharing I was also curious about the training asked by @Santelomar thx!
I will sure explore it carefully once I'll be able to install Musubi-Tuner.

I followed your suggestions and started from fresh clean folder.
I made a venv with **Python 3.12.8** as you suggested.

I copy past your list and tried it via pip install, but I get lots of errors one after another, 
I'm not sure if it's because of order or anything like that, I tried many things but I'm out of ideas.
maybe you can help:

First I realize I have to insert `==` between each name and version like the original requirement format so I added these,
I guess it wasn't an actual requirement list and I didn't realize that that I just copy past, still thanks for sharing 🙏
But that was the most simple error to figure out so far.

Now the normal issues that are always hunting me and probably others:
First it doesn't like the Sageattention folder (yours was G:, I replaced it with my musubi-tune drive and folder)
but it still didn't like it so I had to delete the folder, I'm guessing it shows where it is on your hard drive so I shouldn't put it in the requirement list anyway (I hope I'm not wrong)

Currently the edited  `requirements_5090.txt`  looks like this: 

```
absl-py==2.2.2
accelerate==1.6.0
ascii-magic==2.3.0
av==14.0.1
bitsandbytes==0.45.4
certifi==2025.4.26
charset-normalizer==3.4.1
colorama==0.4.6
contourpy==1.3.2
cycler==0.12.1
diffusers==0.32.1
easydict==1.13
einops==0.7.0
filelock==3.13.1
flash_attn==2.7.4.post1
fonttools==4.57.0
fsspec==2024.6.1
ftfy==6.3.1
grpcio==1.71.0
huggingface-hub==0.30.0
idna==3.10
importlib_metadata==8.7.0
Jinja2==3.1.4
kiwisolver==1.4.8
Markdown==3.8
MarkupSafe==2.1.5
matplotlib==3.10.1
mpmath==1.3.0
networkx==3.3
ninja==1.11.1.4
numpy==2.1.2
opencv-python==4.10.0.84
packaging==25.0
pillow==11.0.0
pip==25.1
protobuf==6.30.2
psutil==7.0.0
pyparsing==3.2.3
python-dateutil==2.9.0.post0
PyYAML==6.0.2
regex==2024.11.6
requests==2.32.3
safetensors==0.4.5
sageattention==2.1.1
setuptools==70.2.0
six==1.17.0
sympy==1.13.3
tensorboard==2.19.0
tensorboard-data-server==0.7.2
tokenizers==0.20.3
toml==0.10.2
torch==2.7.0+cu128
torchaudio==2.7.0+cu128
torchvision==0.22.0+cu128
tqdm==4.67.1
transformers==4.46.3
triton-windows==3.3.0.post19
typing_extensions==4.12.2
urllib3==2.4.0
voluptuous==0.15.2
wcwidth==0.2.13
Werkzeug==3.1.3
wheel==0.45.1
xformers==0.0.30
zipp==3.21.0
```
---

I didn't have any issues with Python 3.10  on the requirements, but I had to try your suggestion since you have it working.

Here is an example of what happens when I try to pip install the list, I also tried pip3 but it's the same errors:

![Image](https://github.com/user-attachments/assets/4adff4ba-867e-42c2-adc4-851061b4e4cf)

So I tried to install SageAttention based on the 3 lines you suggest and also it's on the repo, I know that part worked for me when I was on Python 3.10 but now it's having this issue:

![Image](https://github.com/user-attachments/assets/bac5f804-35bb-4111-9fbc-f84d4e355a89)

Since it didn't allow to install SageAttention and FlashAttention via the requirement_5090.txt list, 
I'm now stuck... again 

I had a feeling it sounds too easy to install hehe, and I CAN'T WAIT to start training a TEST I will 100% read your tips above the moment I'll be able to install Musubi-Tuner... it's still too many errors I can't even understand.

Any suggestions? solutions? 
portable version or one-click install script to help install the impossible musubi-tuner windows relationship will be very helpful.

**Comment by rwfsmith** [comment]
Neither Flash Attention or Sage Attention is required for it to work. I did get the same error at one point. One time I fixed it by updating wheel and setuptools, 

`pip install -U setuptools wheel`

more recently just by running this:

`pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu128 --force -U`

**Comment by df2df** [comment]
Fyi Sage attention messed up training results for me after a few hundred steps when I tried. I could see it happen in the tensorboard during training. Switched back to sdpa and all was good. Could be my install of sage + xformers but I used the same I use in comfy which works fine.

**Comment by AlonDan** [comment]
Thanks @rwfsmith 
I'm still sruggeling with so many errors mostly related to SageAttention and FlashAttention but it did some progress, at least I can install most of the list.

@df2df I knew how to just install successfully without SageAttention / FlashAttention maybe I would, I guess it's recommended or even must have in order to run something in the training, I have no clue because it feels like I'm too far away from actually be able to train anything... fighting with Musubi-Tuner installation is a struggle so far.

I'm so curious try to train something simple and quick already just to see if it works... but I'm still "Mortal Kombat" with Musubi-Tuner installation over here.

**Comment by AlonDan** [comment]
@rwfsmith your help and supportive spirit is very appreciated, thank you so much!

I still don't know if my installation will work, so in order to test I need to setup dataset and other files such as .toml I guess.
So I'm looking at your files and use them as reference / example so I can have something to try.

I'm also making nice batch files so I can always EDIT and ReUSE once it works of course.

---

 I'm trying to understand the Wan Docs, but so far I'm confused with many things that are very new to me so I have many questions, hopefully you can correct me or give me tips based on your experience:

By the order you wrote the above:

- Create Cache:
Can you please explain what exactly this thing do?
I'm not sure if I understood, it took the VIDEO files (in my case, i2v) and extract them to frames or something?
I guess it does some preparation before the training start but I can't follow what exactly it does, it creates temporary files?

---

- Train Lora:
I'm trying to update my folders of course but also, I'm replacing the models to i2v
I'm not sure if BF16 or FP16 I have both, they're 32GB in size.
I just placed ALL models needed inside: 

`D:\AI\Musubi-Tuner-3\models\wan\`

ALL models inside:
```
models_clip_open-clip-xlm-roberta-large-vit-huge-14.pth
models_t5_umt5-xxl-enc-bf16.pth
wan_2.1_vae.safetensors
Wan2.1_VAE.pth   ---  🤔should I get rid of this and use the .safetensors only?
wan2.1_i2v_720p_14B_bf16.safetensors
wan2.1_i2v_720p_14B_fp16.safetensors
```
---

- Convert
I guess when training is done it must convert it so I could test it on ComfyUI.

---

- Regarding video files:
`This was my first lora and it wasn't great, so take this all with a grain of salt, but I focused mostly on training t2v, but it should work similarly for i2v. I used both images and videos, took some 2-4 second clips in .mp4 format, put them into separate folders`

You mention: "took some 2-4 second clips in .mp4 format, put them into separate folders"
- Did you mean each video in a separate folder? or Folder for ALL videos, and Folder for ALL images?

For the quickest train just to see if everything works:
- Can I use 3 (or less?) videos .mp4 ?

Am I in the right direction  with the Folders structures and files:
```
D:\AI\Musubi-Tuner-3\dataset\videos\

0001.mp4
0002.mp4
0003.mp4
```

Resolution: **512x512** for each video
**16 fps** - because if I don't want slow motion, I guess native wan is 16 fps, (I hope I'm not wrong)
about 48 frames / 3 seconds for each video

If I'm correct I also need to place the .txt files in the SAME folder of the videos:
```
0001.txt
0002.txt
0003.txt
```

At the moment I will describe it manually I guess, but later I will check the tool you suggested,
If I understood it's a workflow inside ComfyUI that can actually describe VIDEO files? 
If so, that sounds awesome because I tried something like that only for images.

Do I need a Trigger word for each? is it helpful? is it recommended? 

Do I MUST use .txt files or it's optional ?
If there will be only video files WITHOUT .txt files, will the training IGNORE or CRASH ?

---

- .TOML

This is overwhelming at first glance, because I have NO CLUE what most of these terms means, I'm still trying to follow the Wan Docs but it's not as simple to understand as your examples 👍
My goal for the first lora train is: i2v (image to video) and focus on MOTION so I guess I need to do some specific changes in the .toml to match my video files resolution, fps, length etc.. it's a guess I'm not really sure.

---

Any **rough idea** how long it will take to train something like I mentioned with 3 videos?
You suggested max 10 epochs, once I'll find it I'll change it to 10 and give it a try when everything else will be set.
I believe it's related on many other factors as well, I have no clue... noob-mode.


But I need something ROUGHLY even that the LoRA will not be amazing, just to TEST the TRAINING itself first.


---

If you can please share ANY tips & tricks or guide me what should I do, or NOT I will appreciate it and I will finally be able to try RUN the first TRAIN..... if everything works, I can't tell until I'll set all the files and paths, etc.. that's why I have many questions.

I guess once it will work I'll be able to try different things.

HUGE THANKS! ❤️ for any help, tips, tricks, suggestions, extra reference examples like you did, anything is welcome!

**Comment by Santelomar** [comment]
I just followed the last posts and really good questions I'm also interested in so I will learn how to test.

I did try with Python 3.12.8 as @rwfsmith suggested, it gives so many bugs in my case, it ask for rust, having issues with conflicts with huggingface versions that asks only for Python 3.10
I wonder how you passed all these complicated issues with Python 3.12.8 not that I could install successfully with Python 3.10

basically I still can't run training there are too many issues to even install successfully on Windows and RTX 50xx

@AlonDan did you manage to install successfully? 
also great questions, I hope @rwfsmith or anyone else will share their experience for training i2v I'm also interested in that.

**Comment by AlonDan** [comment]
If anyone still struggle, I finally made a **[GUIDE](https://github.com/kohya-ss/musubi-tuner/issues/273)** on how to install it on Windows 11 + RTX-50xx series.
I hope it helps! 🙏
